### PR TITLE
Understanding  and Implementing Passwordless SSH

### LINKED ISSUE #6276: [Networking]Understanding Passwordless SSH
### Topic Suggestion
Understanding Passwordless SSH

### Writing sample(s): 
[How to Add Pure CSS Waves to Your Website](https://www.section.io/engineering-education/pure-css-wave-animations-website/)

#### Proposed title of the article
Understanding Passwordless SSH

#### Proposed article introduction
Online privacy and security are vital considerations in technology. In the current era, information is comparable to gold. Because of the value that information holds, there has been a rise in better security practices and strategies meant to ensure information remains secure on the internet. The secure shell host protocol (SSH) is among the vital security implementations technology experts use to secure information on transit in an unsecured network. As time passes the cryptographic based protocol has seen a revamp into a passwordless alternative meant to increase its robustness as one of the best security protocols in use. 

#### Key takeaways 
- what is passwordless SSH
- Reasons for preferring passwordless SSH over normal SSH 
- Advantages and disadvantages of passwordless SSH
- Use cases of passwordless SSH
- How to set up passwordless SSH

#### Article quality
Articles on the web only focus on how one can set up passwordless SSH. This article takes a broad but deeper approach to include the advantages and disadvantages of the technology combined with an explanation of the what, why and how of passwordless SSH.
#### References
N/A

### PR DESCRIPTION
**Pull request by lin-chez**
#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [X] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [X] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [X] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #6276

### CODE REVIEW

**Comment by lin-chez** [comment]
@Bayler kindly continue your review on this new PR

**Comment by Bayler** [comment]
@lin-chez Please perform a Grammarly check on your article.

File: content/articles/implementing-and-understanding-passwordless-ssh/index.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+#### 3. Protection from Brute-force Attacks
+A brute-force attack involves an attacker guessing characters to form a password. This technique is practically unattainable on cryptographic-based protocols such as passwordless SSH. Passwordless SSH uses a cryptographic key that makes brute-forcing cumbersome. The cryptographic key is created from mathematical formulas that are easy to compute in one direction but cumbersome to compute in the opposite direction. Thus, once a key is generated, it is cumbersome for an attacker to reverse engineer the generation process. 
+
+#### 4. Helps Reduce Operational Costs in the Long Run
+Passwordless authentication solutions reduce overall security costs. An organization does not need to incur costs from storing passwords on authenticating servers that require maintenance and management. 
+The protocol also frees up the IT department as they will no longer be required to redefine password policies. The IT department will also not need to monitor user activity or detect and prevent password leakages.
+
+#### Disadvantages of Passwordless SSH
+#### 1. Hard to Protect Users after Device Theft
```
Comment (Bayler) [inline content/articles/implementing-and-understanding-passwordless-ssh/index.md:102]: Only the first word is capitalized in a heading, unless it is absolutely necessary. Please go through the whole article and change accordingly

File: content/articles/implementing-and-understanding-passwordless-ssh/index.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+topics: [Security]
+excerpt_separator: <!--more-->
+images:
+
+ - url: /engineering-education/implementing-and-understanding-passwordless-ssh/hero.png
+  alt: Understanding and Implementing Passwordless SSH Example Image
+---
+Online privacy and security are vital considerations in our modern era. The high value of information has caused a need to implement better security practices and strategies. 
+<!--more-->
+The secure shell host protocol (SSH) is one of the important security protocols used to protect data in an insecure network. The cryptographic-based protocol has evolved to a passwordless alternative to increase its resilience.
```
Comment (Bayler) [inline content/articles/implementing-and-understanding-passwordless-ssh/index.md:19]:
```suggestion
The Secure Shell Host protocol (SSH) is one of the important security protocols used to protect data in an insecure network. The cryptographic-based protocol has evolved to a passwordless alternative to increase its resilience.
```

File: content/articles/implementing-and-understanding-passwordless-ssh/index.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+#### What You will learn
+1. What is passwordless SSH?
+2. The working of passwordless SSH.
+3. Its use cases.
+4. Setting up passwordless SSH.
+5. Advantages and disadvantages of passwordless SSH.
+
+#### What is Passwordless SSH
+The Secure Shell or Secure Socket Shell protocol(SSH) is a cryptographic-based network protocol that Yatu Ylonen wrote in 1995. The protocol is applied when establishing secure connections between two devices, i.e., a client and a server. SSH can be implemented using several authentication techniques. The two widely used techniques are public/private-key (cryptographic) based authentication and password-based authentication.
```
Comment (Bayler) [inline content/articles/implementing-and-understanding-passwordless-ssh/index.md:29]:
```suggestion
The Secure Shell or Secure Socket Shell protocol (SSH) is a cryptographic-based network protocol that Yatu Ylonen wrote in 1995. The protocol is applied when establishing secure connections between two devices, i.e., a client and a server. SSH can be implemented using several authentication techniques. The two widely used techniques are public/private-key (cryptographic) based authentication and password-based authentication.
```

Comment (Bayler) [reply content/articles/implementing-and-understanding-passwordless-ssh/index.md:29]: Please add an external link to Yatu Ylonen

File: content/articles/implementing-and-understanding-passwordless-ssh/index.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+2. The server responds with a message encrypted using the received public key from the user. 
+3. The user decrypts the server’s message with its private key. 
+4. The user sends the decrypted message to the server, which ascertains if the message is valid.
+5. If the message matches the one on the server, the server authenticates the user and establishes a secure connection between the two devices. 
+6. Since future connections are automated, users do not repeat the username submission process stated in the first step.
+
+> The lack of passwords is among the main reasons for preferring the passwordless SSH over its password-based alternative.
+
+#### Use Cases of SSH
+The SSH protocol is the remote standard method fo accessing and managing Linux-based servers. Many corporate organizations utilize the protocol to administer and manage their web servers.
```
Comment (Bayler) [inline content/articles/implementing-and-understanding-passwordless-ssh/index.md:48]:
```suggestion
The SSH protocol is the remote standard method for accessing and managing Linux-based servers. Many corporate organizations utilize the protocol to administer and manage their web servers.
```

**Comment by Bayler** [comment]
@lin-chez This is an educative article. Please work on the minor edits before we can move on for final review.

**Comment by ahmadmardeni1** [comment]
Great work team. @lin-chez @Bayler

**Pull request merged**